### PR TITLE
[GECO-99] Giles' admins can now stop listening to Kafka

### PR DESCRIPTION
**Pull request by abhikrish503501**

### CODE REVIEW

File: giles-eco/pom.xml [hunk 1]
```diff
@@ -20,7 +20,9 @@
         <geco.requests.version>0.7</geco.requests.version>
         <geco.util.version>0.4.3</geco.util.version>
         <geco.september.util.version>0.2</geco.september.util.version>
-
+		<geco.kafka-util.version>0.1</geco.kafka-util.version>
```
Comment (jdamerow) [inline giles-eco/pom.xml]: indentation

File: giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <div style="margin-top: 70px;"><center>
+	<c:if test="${listenerStatus}">
+		<form:form
+			action="${pageContext.request.contextPath}/admin/kafka/listeners/stop"
+			method="POST">
+			<input type="submit" value="Stop Kafka Listeners"
+				class="btn btn-lg btn-danger" />
+		</form:form>
+		<p><br>
+        Once Kafka listeners have been stopped, Giles will no longer process image extraction requests.
```
Comment (jdamerow) [inline giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp]: giles does not process image extraction requests

File: giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	<c:if test="${not listenerStatus}">
+		<form:form
+			action="${pageContext.request.contextPath}/admin/kafka/listeners/start"
+			method="POST">
+			<input type="submit" value="Start Kafka Listeners"
+				class="btn btn-lg btn-success" />
+		</form:form>
+		<p><br>
+		Once Kafka listeners have been started, Giles will proceed processing image extraction requests.
```
Comment (jdamerow) [inline giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp]: same here

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <div style="margin-top: 70px;"><center>
+	<c:if test="${listenerStatus}">
+		<form:form
+			action="${pageContext.request.contextPath}/admin/kafka/listeners/stop"
+			method="POST">
+			<input type="submit" value="Stop Kafka Listeners"
+				class="btn btn-lg btn-danger" />
+		</form:form>
+		<p><br>
+        Kafka listeners have been started!!!
```
Comment (jdamerow) [inline giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp]: one exclamation mark is more than enough ;)

File: giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+	<c:if test="${not listenerStatus}">
+		<form:form
+			action="${pageContext.request.contextPath}/admin/kafka/listeners/start"
+			method="POST">
+			<input type="submit" value="Start Kafka Listeners"
+				class="btn btn-lg btn-success" />
+		</form:form>
+		<p><br>
+		Kafka listeners have been stopped!!!
```
Comment (jdamerow) [inline giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp]: same here.

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: giles-eco/pom.xml [hunk 6]
```diff
@@ -20,7 +20,9 @@
         <geco.requests.version>0.7</geco.requests.version>
         <geco.util.version>0.4.3</geco.util.version>
         <geco.september.util.version>0.2</geco.september.util.version>
-
+        <geco.kafka-util.version>0.1</geco.kafka-util.version>
+        <geco.september.util.version>0.2</geco.september.util.version>
```
Comment (jdamerow) [inline giles-eco/pom.xml]: september util version is already there

File: giles-eco/src/main/webapp/WEB-INF/tiles-defs.xml [hunk 7]
```diff
@@ -95,7 +95,9 @@
     <definition name="admin/requests/inProgress" extends="base.definition">
         <put-attribute name="content" value="/WEB-INF/views/admin/requests_inprogress.jsp" />
     </definition>
-	
+    <definition name="admin/kafka/listeners" extends="base.definition">
+		<put-attribute name="content" value="/WEB-INF/views/admin/manage_listeners.jsp" />
+	</definition>
```
Comment (jdamerow) [inline giles-eco/src/main/webapp/WEB-INF/tiles-defs.xml]: formatting

File: giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    <div style="margin-top: 70px;"><center>
+	<c:if test="${listenerStatus}">
+		<form:form
+			action="${pageContext.request.contextPath}/admin/kafka/listeners/stop"
+			method="POST">
+			<input type="submit" value="Stop Kafka Listeners"
+				class="btn btn-lg btn-danger" />
+		</form:form>
+		<p><br>
+        Kafka listeners have been started !
```
Comment (jdamerow) [inline giles-eco/src/main/webapp/WEB-INF/views/admin/manage_listeners.jsp]: remove blanks

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: giles-eco/src/main/webapp/WEB-INF/tiles/skeleton.jsp [hunk 9]
```diff
@@ -111,6 +111,9 @@
           	<li role="presentation">
           		<a href="<c:url value="/admin/apps" />" >Apps</a>
           	</li>
+          	<li role="presentation">
+          		<a href="<c:url value="/admin/kafka/listeners" />">Kafka</a>
+          	</li>
```
Comment (jdamerow) [inline giles-eco/src/main/webapp/WEB-INF/tiles/skeleton.jsp]: It get's a little crowded up there in the main menu. Let's move this as a new entry under "Requests" as "Start/Stop Kafka Listeners".

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: giles-eco/src/main/webapp/WEB-INF/tiles/skeleton.jsp (hunk 9 above)
Comment (abhikrish503501) [reply giles-eco/src/main/webapp/WEB-INF/tiles/skeleton.jsp]: Made the changes as described above and requested for the review.

**Pull request merged**